infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesPods.java

```diff
@@ -257,16 +258,64 @@ public void onClose(KubernetesClientException cause) {
       if (pod == null) {
         podRunningFuture.completeExceptionally(
             new InfrastructureException("Specified pod " + name + " doesn't exist"));
-      }
-      if (predicate.test(pod)) {
-        podRunningFuture.complete(null);
+      } else {
+        handleStartingPodStatus(podRunningFuture, pod);
       }
     } catch (KubernetesClientException | InfrastructureException ex) {
       podRunningFuture.completeExceptionally(ex);
     }
     return podRunningFuture;
   }
 
+  /**
+   * Complete the future of the pod depending on it's status:
+   *
+   * <ul>
+   *   <li>complete successfully in case of "Running" pod state.
```

Comment:
It's cool to have such javadoc on private method, but I prefer to have this one on public one `waitRunningAsync` to make clear for clients what and when will happen.